### PR TITLE
[manuf] Update pre-cert-endorsment perso extension hook

### PR DESCRIPTION
**Pull request by tommychiu-github**
Add one more parameter to the hook for accessing flash device state.

### CODE REVIEW

File: sw/device/silicon_creator/manuf/base/ft_personalize.c [hunk 1]
```diff
@@ -523,7 +523,8 @@ static status_t personalize_endorse_certificates(ujson_t *uj) {
  */
 extern status_t personalize_extension_pre_cert_endorse(
```
Comment (vbendeb) [inline sw/device/silicon_creator/manuf/base/ft_personalize.c:524]: this is a question for @timothytrippel I guess - why are these externs not in a .h file? 

I got burnt by the linker happily linking external modules which had these functions but a misplaced parameter order.

Comment (timothytrippel) [reply sw/device/silicon_creator/manuf/base/ft_personalize.c:524]: Unfortunately, this was a result of bit rot / several refactorings. I agree, we can move this to a header file, but that can be done in a follow up PR.

**Review by timothytrippel — APPROVED** [review]

File: sw/device/silicon_creator/manuf/base/tpm_personalize_ext.c [hunk 2]
```diff
@@ -135,7 +135,8 @@ static status_t personalize_gen_tpm_certificates(
 
 status_t personalize_extension_pre_cert_endorse(
     ujson_t *uj, manuf_certgen_inputs_t *certgen_inputs,
-    manuf_certs_t *tbs_certs, cert_flash_info_layout_t *cert_flash_layout) {
+    manuf_certs_t *tbs_certs, cert_flash_info_layout_t *cert_flash_layout,
+    dif_flash_ctrl_state_t *flash_ctrl_handle) {
```
Comment (timothytrippel) [inline sw/device/silicon_creator/manuf/base/tpm_personalize_ext.c:139]: You will need to `#include "sw/device/lib/dif/dif_flash_ctrl.h"` at the top for this to compile. Same for the other extension files here.

**Review by timothytrippel — APPROVED** [review]

File: sw/device/silicon_creator/manuf/extensions/BUILD.bazel [hunk 3]
```diff
@@ -28,6 +28,7 @@ cc_library(
         "@//sw/device/lib/testing/test_framework:status",
         "@//sw/device/lib/testing/test_framework:ujson_ottf",
         "@//sw/device/silicon_creator/lib/cert",
+        "@//sw/device/silicon_creator/manuf/lib:flash_info_fields",
```
Comment (timothytrippel) [inline sw/device/silicon_creator/manuf/extensions/BUILD.bazel:31]: Looks like you included the wrong dep here. You should be including `@//sw/device/lib/dif:flash_ctrl`: https://github.com/lowRISC/opentitan/blob/34b8fc33e36b3ed2a32e6cd50b451a47f16b202d/sw/device/lib/dif/BUILD#L324

Same below.

File: sw/device/silicon_creator/manuf/extensions/BUILD.bazel [hunk 4]
```diff
@@ -39,6 +40,7 @@ cc_library(
...
```
Comment (timothytrippel) [inline sw/device/silicon_creator/manuf/extensions/BUILD.bazel:43]: See above comment.

File: sw/device/silicon_creator/manuf/base/BUILD [hunk 5]
```diff
@@ -265,6 +265,7 @@ cc_library(
         "//sw/device/silicon_creator/lib/cert:tpm_ek_template_library",
         "//sw/device/silicon_creator/lib/drivers:flash_ctrl",
         "//sw/device/silicon_creator/lib/drivers:hmac",
+        "//sw/device/silicon_creator/manuf/lib:flash_info_fields",
```
Comment (timothytrippel) [inline sw/device/silicon_creator/manuf/base/BUILD:268]: you don't need this dep. you only need the `"//sw/device/lib/dif:flash_ctrl",` which is already included above.

**Review by timothytrippel — APPROVED** [review]

**Pull request merged**